### ISSUE TITLE
Alternate Frequency Corpuses - Suggestions welcome

### ISSUE THREAD

**Issue by sschmidTU** [sschmidTU/anki-frequency-inserter#3]
InnocentCampus is great, but with sourcing ~5000 novels, it's a little small (though mostly sufficient), and specific to books.
Alternatives would be sources using News, Movies, Wikipedia, Twitter, etc.
Ideally, we would be able to use any of these frequencies, by user choice (or insert all of them into different fields).

There's a popular Anime/J-Drama corpus -> research. (see next comment)
(It ranks words differently though, 1 = most common word, 2 = 2nd most common word, etc. - inverse relationship to InnocentCampus frequency, which is number of occurences. This actually lead to confusions in forums)

It's easy enough to exchange the corpus, right now it's just a Javascript object (Hashmap) called `innocent_terms_complete` inserted as a global variable through a `<script>` tag in the index.html.
It was generated via `tools/parseCorpus.js`.
So currently `frequency = innocent_terms_complete[word]`. (more or less)

_Note for Latin nerds: the Latin plural is corpora, but corpuses is also an allowed plural in English. I love Latin, but English is not Latin._
<sub><sup>Also, the plural of octopus is octopuses, not octopi, because it's a Greek word, not Latin, the Greek plural being octopodes. But the dictionary is generous and accepts all 3, reflecting common usage.</sup></sub>

**Comment by sschmidTU** [comment]
Frequency Corpuses research (WIP):

* The Anime/JDrama frequency corpus may be this one, using ~12000 Anime+Drama subtitle files:
https://github.com/chriskempson/japanese-subtitles-word-frequency-list

This one actually has both number of occurences and frequency ranking ("nth most common word"), which is nice.

* There's also one using ~200 anime shows, which is too small a sample for my tastes, but has some interesting findings ("the top ~900 kanji make 90% of kanji occurences, top ~1900 make 99%"):
https://www.reddit.com/r/LearnJapanese/comments/crlsqj/googlesheet_anime_frequency_list/
<img src="https://user-images.githubusercontent.com/33069673/132875655-2c7eaecc-202d-4103-8f83-ccf5ebae61c5.png" height="300">

**Comment by patarapolw** [comment]
I've just found another resource on PyPI ([toiro](https://pypi.org/project/toiro/)).

```py
from toiro import datadownloader, tokenizers

# A list of avaliable corpora in toiro
corpora = datadownloader.available_corpus()
print(corpora)
# => ['livedoor_news_corpus', 'yahoo_movie_reviews', 'amazon_reviews', 'chABSA_dataset']

available_tokenizers = tokenizers.available_tokenizers()
print(available_tokenizers.keys())
# => dict_keys(['nagisa', 'janome', 'mecab-python3', 'sudachipy', 'spacy', 'ginza', 'kytea', 'jumanpp', 'sentencepiece', 'fugashi-ipadic', 'tinysegmenter', 'fugashi-unidic'])
```

I am not really sure if segmenters are required, but I added them just in case.

**Comment by sschmidTU** [comment]
BCCWJ corpus is now added (long version, separate URL for now, 5.8MB download):
https://sschmidtu.github.io/anki-frequency-inserter/index_BCCWJ.html?expressionFieldName=Expression&frequencyFieldName=FrequencyBCCWJ

main commit: b64a447bd9ffaccd9b1b8bdebb4ad2d8d0c237ed
unification: 36aba625f0a49901d74f15be57cb2ab0478d983c

This is the *Balanced Corpus of Contemporary Written Japanese*, which uses relative frequency (100 = 100th most common word) instead of absolute frequency like InnocentCorpus (100 = occurs 100 times in these 5000 books).
(*we could also convert InnocentCorpus to relative frequency via code if desired by the user*)

In my experience, both corpuses have some interesting differences and common words missing, so they supplement each other very well in my Anki cards.
I wrote more about it in [a post on Wanikani](https://community.wanikani.com/t/anki-word-frequency-inserter-learn-most-common-words-first/53407/23).

@patarapolw tagging you in case you're interested.

Still open for more corpus suggestions!
(*The ones from PyPI sound interesting, just didn't get to take a look yet*)

Currently, the BCCWJ corpus just needs its own `index_BCCWJ.html` (and corpus `terms_BCCWJ.js`), the main code is now [unified](https://github.com/sschmidTU/anki-frequency-inserter/commit/36aba625f0a49901d74f15be57cb2ab0478d983c) in `frequencyInserter.js` and can be easily expanded for more corpuses.

We could unify the HTML into one page as well, but then we'd need radio buttons to choose the corpus,
and more importantly to load the corpus after page load on user click, so that the user doesn't have to download all corpuses at once. (Innocent is ~1.7MB zipped, BCCWJ ~5.8MB)
This might take longer for the user, and we'd need to dynamically `require` the corpus `.js` or something.

**Comment by musjj** [comment]
Would appreciate it if you can also add the JPDB frequency dictionary:

https://github.com/Kuuuube/yomitan-dictionaries

**Comment by sschmidTU** [comment]
@musjj Yes, that would be great, I love jpdb, and they seem to have a very good corpus for frequencies.
Will see if I can integrate this soon, hope I find the time :)
The only question is if there might be some licensing issue there, using their frequencies. Though I'd hope they see that as free advertisement, and they haven't done anything against that repo you linked since 11 months.

**Comment by musjj** [comment]
There's also https://jiten.moe which provides an official downloadable frequency list which is kept up-to-date: https://jiten.moe/other. Their catalog is generally bigger compared to jpdb, but they have slightly fewer novels. It's constantly being updated though, so that might change in the future.
